usaspending_api/bulk_download/v2/views.py

```diff
@@ -0,0 +1,311 @@
+            # which will eventually run csv_selection.write_csvs(**kwargs)
+            # (see generate_bulk_zip.py)
+            message_attributes = {
+                'download_job_id': {
+                    'StringValue': str(kwargs['download_job'].bulk_download_job_id),
+                    'DataType': 'String'
+                },
+                'file_name': {
+                    'StringValue': kwargs['file_name'],
+                    'DataType': 'String'
+                },
+                'columns': {
+                    'StringValue': json.dumps(kwargs['columns']),
+                    'DataType': 'String'
+                },
+                'sources': {
+                    'StringValue': json.dumps(tuple([source.toJsonDict() for source in kwargs['sources']])),
+                    'DataType': 'String'
+                }
+            }
+            queue.send_message(MessageBody='Test', MessageAttributes=message_attributes)
+
+        return self.get_download_response(file_name=timestamped_file_name)
+
+
+def verify_requested_columns_available(sources, requested):
+    bad_cols = set(requested)
+    for source in sources:
+        bad_cols -= set(source.columns(requested))
+    if bad_cols:
+        raise InvalidParameterException('Unknown columns: {}'.format(bad_cols))
+
+
+class BulkDownloadListAgenciesViewSet(APIView):
+    def post(self, request):
+        """Return list of agencies if no POST data is provided.
+        Otherwise, returns sub_agencies/federal_accounts associated with the agency provided"""
+        response_data = {'agencies': [],
+                         'sub_agencies': [],
+                         'federal_accounts': []}
+
+        agency_id = None
+        post_data = request.data
+        if post_data:
+            if 'agency' not in post_data:
+                raise InvalidParameterException('agency parameter not provided')
+            agency_id = post_data['agency']
+
+        # Get all the top tier agencies
+        toptier_agencies = list(ToptierAgency.objects.all().values('name', 'toptier_agency_id', 'cgac_code'))
+
+        if not agency_id:
+            # Return all the agencies if no agency id provided
+            response_data['agencies'] = toptier_agencies
+        else:
+            # Get the top tier agency object based on the agency id provided
+            top_tier_agency = list(filter(lambda toptier: toptier['toptier_agency_id'] == agency_id, toptier_agencies))
+            if not top_tier_agency:
+                raise InvalidParameterException('Agency ID not found')
+            top_tier_agency = top_tier_agency[0]
+            # Get the sub agencies and federal accounts associated with that top tier agency
+            response_data['sub_agencies'] = Agency.objects.filter(toptier_agency_id=agency_id)\
+                .annotate(subtier_agency_name=F('subtier_agency__name'),subtier_agency_id=F('subtier_agency__subtier_agency_id'))\
```

Comment:
If all you need to do is change the column names returned, you can perform the same operation in .values()

Reply:
Oh missed that. Fixed.